Dev

vibriss opened the pull request:
init models

vibriss committed:
init models

vibriss committed:
init models

vibriss committed:
init database

ramilmmk commented on pom.xml:
```diff
@@ -35,6 +73,17 @@
+		</dependency>
+		<dependency>
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
```
используется и h2 и postgres? или что-то лишнее?

ValeriyEmelyanov replied:
обсуждали, что для продуктива будет postgres, для тестирования h2

ramilmmk commented on src/main/java/com/training/courier/model/Courier.java:
```diff
@@ -0,0 +1,51 @@
+     * Name
+     */
+    @Column(name = "name")
+    private String name;
```
Курьер всё-таки человек, давайте хотя бы добавим ему фамилию, имя и возраст

vibriss replied:
Ок

ramilmmk commented on src/main/java/com/training/courier/model/Salary.java:
```diff
@@ -0,0 +1,52 @@
+@Entity
+@Table(name = "salary")
+@Data
+public class Salary {
```
Я б добавил какие-то показатели для расчёта зп(например, количество доставок, количество отработанных часов....) + указывать за какой период эта зп

vibriss replied:
Сходу не придумали, как сделать расчёт зарплаты курьера, поэтому оставили на потом. Это предварительная модель, которая точно поменяется.

ramilmmk commented on src/main/resources/db/migration/V1_0_0_1__init_script.sql:
```diff
@@ -0,0 +1,59 @@
+    city                          varchar(32)    NOT NULL,
+    street                       varchar(32),
+    house                       varchar(4)      NOT NULL,
+    apartment                 varchar(4),
```
region (32) - Республика Северная Осетия - Алания - сразу вызывает ошибку
street (32) - улица "Территория, изъятая из земель подсобного хозяйства Всесоюзного центрального совета профессиональных союзов для организации крестьянского хозяйства" - сразу вызывает ошибку
house (4) - 71к4Г-Дс5 - сразу вызывает ошибку
app (4) - думаю тут тоже стоит увеличить

ramilmmk commented on src/main/java/com/training/courier/model/Client.java:
```diff
@@ -0,0 +1,51 @@
+     * Name
+     */
+    @Column(name = "name")
+    private String name;
```
Что такое name у клиента? Название клиента? Клиент - человек, курьер должен обращаться к клиенту по ФИО.

ValeriyEmelyanov commented on src/main/java/com/training/courier/model/Cargo.java:
```diff
@@ -0,0 +1,63 @@
+@Data
+public class Cargo {
+
+    /**
+     * Unique identifier
+     */
+    @Column(name = "id")
+    private Long id;
```
Должна быть аннотация @Id. Первичный ключ должен быть, просто он не будет автоматически генерироваться. Для остальных сущностей тоже.

vibriss replied:
Ок

ValeriyEmelyanov commented on src/main/java/com/training/courier/model/Delivery.java:
```diff
@@ -0,0 +1,93 @@
+    /**
+     * Courier information
+     */
+    @OneToOne(cascade = CascadeType.PERSIST)
+    @JoinColumn(name = "courier_id", referencedColumnName = "id")
+    private Courier courier;
```
ManyToOne, у курьера много доставок. Предлагаю в эту сущность добавить trackingNumber для сопоставления с Delivery из core и для записи в трегинг.

ValeriyEmelyanov commented on src/main/java/com/training/courier/model/Delivery.java:
```diff
@@ -0,0 +1,93 @@
+@Entity
+@Table(name = "delivery")
+@Data
+public class Delivery {
```
Может так и назвать CourierDeliveri, чтобы не путаться.

ValeriyEmelyanov commented on src/main/java/com/training/courier/model/Delivery.java:
```diff
@@ -0,0 +1,93 @@
+     * Enable to notify client about cargo  //TODO определиться с необходимостью этого поля
+     */
+    @Column(name = "enabled_notifications")
+    private Boolean enabledNotifications;
```
скорее нет

ValeriyEmelyanov commented on src/main/resources/db/migration/V1_0_0_1__init_script.sql:
```diff
@@ -0,0 +1,59 @@
+CREATE TABLE IF NOT EXISTS address (
+    id                             bigserial          NOT NULL PRIMARY KEY,
```
насколько знаю достаточно `id bigserial PRIMARY KEY`, без `NOT NULL`, postgress не позволит NULL в первичном ключе. (и далее тоже)

vibriss committed:
changes in model and db init script

vibriss commented on src/main/resources/db/migration/V1_0_0_1__init_script.sql:
```diff
@@ -1,47 +1,48 @@
-    apartment                 varchar(4),
-    created                     TIMESTAMP     NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    id                             bigserial          PRIMARY KEY,
+    region                       varchar(255)   NOT NULL,
```
@ValeriyEmelyanov, я спрашивал, на каком размере полей остановиться

ramilmmk replied:
По-моему, допустимо будет сделать так:
region 128
city 128
street 255
house 16
appartment 16

vibriss committed:
init openFeign client
delivery controller interface
init dto
service interface
urls interface
nonNullApi
javadoc practice

vibriss committed:
minor changes

vibriss committed:
minor changes

ValeriyEmelyanov commented on src/main/java/com/training/courier/Urls.java:
```diff
@@ -0,0 +1,11 @@
+
+public interface Urls {
+
+    String ROOT = "/";
```
Тут есть мой косяк. ROOT не может быть в текущей реализации "/", тогда в полном маршруте получается два слеша. Сделай ROOT = "".

vibriss replied:
Да, я видел этот комментарий в Core, у себя уже переделал, в следующем коммите это будет.

vibriss committed:
added more endpoints
